File: Polimorfismo/Ejercicio-1/Videojuego.py
class Videojuego:
    def __init__(self, nombre, plataforma, cantidad_jugadores=None):
        self.nombre = nombre
        self.plataforma = plataforma
        if cantidad_jugadores is None:
            self.cantidad_jugadores = 2  # Valor por defecto
        else:
            self.cantidad_jugadores = cantidad_jugadores

    def mostrar(self):
        print(f"Nombre: {self.nombre}")
        print(f"Plataforma: {self.plataforma}")
        print(f"Jugadores: {self.cantidad_jugadores}")

    def agregar_jugador(self, cantidad=1):
        self.cantidad_jugadores += cantidad


if __name__ == "__main__":
    v1 = Videojuego("It takes two", "PC")
    v2 = Videojuego("FIFA 24", "PS5", 5)

    print("----------ANTES DE AGREGAR JUGADORES----------")
    v1.mostrar()
    v2.mostrar()

    v1.agregar_jugador()
    v2.agregar_jugador(3)

    print("----------DESPUÉS DE AGREGAR JUGADORES--------")
    v1.mostrar()
    v2.mostrar()

File: Polimorfismo/Ejercicio-1/Videojuego.java
public class Videojuego {
    private String nombre, plataforma;
    private int cantidadJugadores;
    
    public Videojuego(String nombre, String plataforma)
    {
        this.nombre=nombre;
        this.plataforma=plataforma;
        this.cantidadJugadores=2;
    }
    
    public Videojuego(String nombre, String plataforma, int cantidadJugadores)
    {
        this.nombre=nombre;
        this.plataforma=plataforma;
        this.cantidadJugadores=cantidadJugadores;
    }
    
    public void mostrar()
    {
        System.out.println("Nombre: "+nombre);
        System.out.println("Plataforma: "+plataforma);
        System.out.println("Jugadores: "+cantidadJugadores);
    }
    
    public void agregarJugador()
    {
        this.cantidadJugadores+=1;
    }
    
    public void agregarJugador(int cantidad)
    {
        this.cantidadJugadores+=cantidad;
    }
    
    public static void main(String[] args) {
        // TODO code application logic here
        Videojuego v1=new Videojuego("It takes two", "PC");
        Videojuego v2=new Videojuego("FIFA 24", "PS5", 5);
        
        System.out.println("----------ANTES DE AGREGAR JUGADORES----------");
        v1.mostrar();
        v2.mostrar();
        
        v1.agregarJugador();
        v2.agregarJugador(3);
        
        System.out.println("----------DESPUES DE AGREGAR JUGADORES--------");
        v1.mostrar();
        v2.mostrar();
    }  
}